### PR TITLE
[TensorIR][M1c] LCA detector

### PR DESCRIPTION
**Pull request by Hzfengsy**
Note that the LCA detector is a helper analysis for lowering, which can help find where the buffer should be allocated.

cc @tqchen @junrushao1994 @jroesch @comaniac

### CODE REVIEW

File: src/tir/analysis/buffer_access_lca_detector.cc [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+#include <tvm/tir/analysis.h>
+#include <tvm/tir/stmt_functor.h>
+
+#include "../../support/arena.h"
+
+namespace tvm {
+namespace tir {
+
+/*!
+ * \brief Detect the LCA position of Buffer access.
```
Comment (xqdan) [inline src/tir/analysis/buffer_access_lca_detector.cc:34]: LCA = LowestCommonAncestor, or life-cycle analysis？ Add the full name of LCA would be better.

File: python/tvm/tir/analysis/analysis.py [hunk 2]
```diff
@@ -16,9 +16,14 @@
 # under the License.
 """Wrapping existing analysis utils."""
 # pylint: disable=invalid-name
-
+from __future__ import annotations
```
Comment (junrushao1994) [inline python/tvm/tir/analysis/analysis.py:19]: I hit an issue a while ago that the current TVM version probably didn't support this :-(

**Review by junrushao1994 — APPROVED** [review]
LGTM. I don't have other comments :-)

File: tests/python/unittest/test_tir_analysis_detect_buffer_access_lca.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+                )
+
+    for i, j in tir.grid(16, 16):
+        with tir.block([16, 16]) as [vi, vj]:
+            tir.bind(vi, i)
+            tir.bind(vj, j)
+            C[vi, vj] = B[vi, vj]
+
+
+def test_buffer_laod_store():
```
[the remark this answers is not in the document]
Comment (Hzfengsy) [reply tests/python/unittest/test_tir_analysis_detect_buffer_access_lca.py:67]: Thanks for pointing out the typo.

File: src/tir/analysis/buffer_access_lca_detector.cc (hunk 1 above)
Comment (Hzfengsy) [reply src/tir/analysis/buffer_access_lca_detector.cc:34]: Great suggestion!

**Review by xqdan — APPROVED** [review]

**Review by MasterJH5574 — APPROVED** [review]

**Review by jcf94 — APPROVED** [review]

**Comment by tqchen** [comment]
@Hzfengsy please fix the lint error

**Pull request merged**